src/ConcreteOperations/linear_combination.jl

```diff
@@ -0,0 +1,29 @@
+export linear_combination
+
+"""
+    linear_combination(P1::SimpleSparsePolynomialZonotope, P2::SimpleSparsePolynomialZonotope)
+
+compute the linear combination of simple sparse polynomial zonotopes `P1` and `P2`.
+
+### Input
+
+- `P1` -- simple sparse polynomial zonotope
+- `P2` -- simple sparse polynomial zonotope
+
+### Output
+
+linear combination of `P1` and `P2`.
+"""
+function linear_combination(P1::SimpleSparsePolynomialZonotope, P2::SimpleSparsePolynomialZonotope)
+    c = 0.5 * (center(P1) + center(P2))
+    G = 0.5 * hcat(center(P1) - center(P2), genmat(P1), genmat(P1), genmat(P2), -genmat(P2))
+    E1, E2 = expmat(P1), expmat(P2)
+    et = promote_type(eltype(E1), eltype(E2))
+    E = hcat(vcat(zeros(et, nparams(P1) + nparams(P2)), one(et)),
+             vcat(E1, zeros(et, nparams(P2) + 1, ngens(P1))),
+             vcat(E2, zeros(et, nparams(P2), ngens(P1)), ones(et, 1, ngens(P1))),
+             vcat(zeros(et, nparams(P1), ngens(P2)), E2, zeros(et, 1, ngens(P2))),
+             vcat(zeros(et, nparams(P1), ngens(P2)), E2, ones(et, 1, ngens(P2))))
```

Comment:
Hm, this does not exactly look "sparse" :sweat_smile:

Reply:
actually it seems that concatenating a sparse matrix with a dense matrix is "smart enough" to avoid converting to dense and storing zeros

```julia
julia> c = [2.0, 0.0]
2-element Vector{Float64}:
 2.0
 0.0

julia> G = [1 2;2 2.]
2×2 Matrix{Float64}:
 1.0  2.0
 2.0  2.0

julia> E = SparseMatrixCSC([1 4;1 2])
2×2 SparseMatrixCSC{Int64, Int64} with 4 stored entries:
 1  4
 1  2

julia> S = SimpleSparsePolynomialZonotope(c, G, E)
SimpleSparsePolynomialZonotope{Float64, Vector{Float64}, Matrix{Float64}, SparseMatrixCSC{Int64, Int64}}([2.0, 0.0], [1.0 2.0; 2.0 2.0], sparse([1, 2, 1, 2], [1, 1, 2, 2], [1, 1, 4, 2], 2, 2))

julia> P = linear_combination(S, S)
SimpleSparsePolynomialZonotope{Float64, Vector{Float64}, Matrix{Float64}, SparseMatrixCSC{Int64, Int64}}([2.0, 0.0], [0.0 0.5 … -0.5 -1.0; 0.0 1.0 … -1.0 -1.0], sparse([5, 1, 2, 1, 2, 1, 2, 5, 1, 2  …  3, 4, 3, 4, 3, 4, 5, 3, 4, 5], [1, 2, 2, 3, 3, 4, 4, 4, 5, 5  …  6, 6, 7, 7, 8, 8, 8, 9, 9, 9], [1, 1, 1, 4, 2, 1, 1, 1, 4, 2  …  1, 1, 4, 2, 1, 1, 1, 4, 2, 1], 5, 9))

julia> expmat(P)
5×9 SparseMatrixCSC{Int64, Int64} with 21 stored entries:
 ⋅  1  4  1  4  ⋅  ⋅  ⋅  ⋅
 ⋅  1  2  1  2  ⋅  ⋅  ⋅  ⋅
 ⋅  ⋅  ⋅  ⋅  ⋅  1  4  1  4
 ⋅  ⋅  ⋅  ⋅  ⋅  1  2  1  2
 1  ⋅  ⋅  1  1  ⋅  ⋅  1  1
```

but I definitely agree the code is unidiomatic at best, not sure what's the best way to write it. Maybe allocate the finale exponents matrix first and then fill the non-zero elements?

Reply:
I do not know. Typically you can outsource to a helper function and add a method for sparse matrices.

Reply:
But then again, this is probably not so important. We can keep it for now. I was just concerned that you would get a dense matrix, but you showed that this does not happen.

Reply:
Just for the record, the implementation looks correct :+1: